src/crimson/os/seastore/transaction_manager.h

```diff
@@ -194,6 +194,30 @@ class TransactionManager : public ExtentCallbackInterface {
     }
   }
 
+  template <typename T>
+  base_iertr::future<TCachedExtentRef<T>> read_cached_pin(
+    Transaction &t,
+    LBAMappingRef pin)
+  {
+    auto v = pin->get_logical_extent(t);
+    if (v.has_child()) {
+      return v.get_child_fut().safe_then([pin=std::move(pin)](auto extent) {
+#ifndef NDEBUG
+        auto lextent = extent->template cast<LogicalCachedExtent>();
+        auto pin_laddr = pin->get_key();
+        if (pin->is_indirect()) {
+          pin_laddr = pin->get_intermediate_base();
+        }
+        assert(lextent->get_laddr() == pin_laddr);
+#endif
+	return extent->template cast<T>();
+      });
+    } else {
+      return base_iertr::make_ready_future<TCachedExtentRef<T>>(
+	TCachedExtentRef<T>());
+    }
+  }
```

Comment:
👍 Can we step further to fully replace `Cache::get_extent_if_cached()` (there are 2 places) by `TM::get_extent_if_cached()` (seems the original name looks more accurate ?), in a separate commit?

Reply:
It seems that the other usage of `Cache::get_extent_if_cached()` is to search through the Cache by the paddr provided by the `BackrefManager`, and there's no `LBAMapping` to follow. I think we can drop it in the PR that change the type of `Cache::extents` from `ExtentIndex` to `std::list`, what do you think?

Reply:
> Make sense to me.
> 
> Further comment:
> 
> * Can `read_cached_pin` be renamed to e.g. `get_extent_if_linked`? Because `read_cached_pin` doesn't involve reading.
> * Looks `read_cached_pin` can be reused by `read_pin()` directly.
> 

Yeah, will do.

> Or, looks the implementation of `read_cached_pin` can be encapsulated inside the mapping implementation, so that it can be reused by `read_pin()`, `read_pin_by_type()` and `remap_pin()`.